Documentation/release-guide.md

```diff
@@ -4,12 +4,76 @@ This document details the strategy for releases and the process to manage them.
+
+### Preview Releases
+
+When work starts on new major versions, that are not going to be 100% compatible, or will have fundamental changes, then they will be released as previews, and the packages will be marked as preview in nuget. For example 2.0 work will be released as a 2.0 preview, rather than as a 1.x.
+
+### .NET Versions
+
+Support for specific versions of the .NET runtime in Core WCF will be timeboxed by the support durations for those runtimes by Microsoft. For example .NET 5 will be supported until May 8 2022 (based on dates from [here](https://docs.microsoft.com/en-us/lifecycle/products/microsoft-net-and-net-core)). 
+
+For engineering reasons, major releases of CoreWCF may drop support for older versions of .NET runtime. This is so that Core WCF can take advanatge of runtime features that are not present in the older runtime. For example, the current plan is that shortly after 1.0 is shippped, work will start on 2.0 which will drop .NET Framework and .NET Core 3.1 support, this is because we need to take a dependency on newer ASP.NET core features that are not available down level.
+
+## Microsoft support
+
+Support for CoreWCF will be available from Microsoft Support with the following conditions:
+
+- CoreWCF is a subset of WCF. Support is not intended to provide or assist with missing functionality found during app migration from WCF or in new development.
+
+- OS Platform support for Core WCF matches OS platform support for the underlying versions of .NET or .NET Core. For e.g. if Core WCF 1.0 supports .NET 5.0 then Core WCF will support the same OS platforms as .NET 5.0 as long as those underlying OS versions are themselves in support by the OS vendor.  
+
+- CoreWCF has major and minor versions. Major versions can take breaking changes, minor versions will not (and we will fix accidental breaks in minor versions.)
+  - E.g. 1.0 -> 2.0 is a major version update with breaking changes while 2.0 -> 2.1 is a minor update and no intentional breaking changes.
+
+- Major versions will declare what .NET runtime & platform is supported and may drop support for an older .NET runtime that is still under support. 
+  - E.g. CoreWCF 1.0 may support .NET Core 3.1 and .NET 5.0 while CoreWCF 2.0 may choose to only support .NET 6.0.
+
+- Minor versions always support the same .NET runtime & platform as the parent major, they do not drop support for a .NET runtime that was supported by the parent major as long as the .NET runtime & ASP.NET Core itself is still in support. 
+  - E.g. If CoreWCF  1.0 supported .NET Core 3.1 then Core WCF 1.1 will also support .NET Core 3.1 as long as .NET Core 3.1 remains in support. Core WCF 2.x will follow whatever was declared supported when 2.0 shipped. 
+
+- Minor versions will be supported for 6 months after the successor ships. 
```

Comment:
We're missing verbiage on how long major versions will be supported. After 3.0 is released, how long will we support 2.x? I think that should be called out more clearly here.